lib/gollum-lib/file.rb

```diff
@@ -47,18 +36,27 @@ def path_compare(query, match_path, hyphened_tags, case_insensitive)
     # that if you specify try_on_disk=true, you may or may not get a file
     # for which on_disk? is actually true.
     def self.find(wiki, path, version, try_on_disk = false, global_match = false)
-      map = wiki.tree_map_for(version.to_s)
-
       query_path = Pathname.new(::File.join(['/', wiki.page_file_dir, path].compact)).cleanpath.to_s
-      query_path.sub!(/^\/\//, '/') if Gem.win_platform? # On Windows, Pathname#cleanpath will leave double slashes at the start of a path intact, so sub them out.
+      query_path.sub!(/^\/+/, '') # On Windows, Pathname#cleanpath will leave double slashes at the start of a path, so replace all (not just the first) leading slashes
+      segments = query_path.split('/')
+      filename = segments.last
+      dir = segments[0..-2].join('/')
 
-      begin
-        entry = map.detect do |entry|
-          path_match(query_path, entry, global_match, wiki.hyphened_tag_lookup, wiki.case_insensitive_tag_lookup)
+      if global_match
+        return nil
```

Comment:
This still needs to be implemented. My proposal would be to just use the old method (with the treemap) for this. Since global tag lookup  is a legacy support feature (although  I assume some people use it because they like it), it's ok if that has the slow performance.